test/assignment1/username-from-conf-file.h

```diff
@@ -1,34 +1,45 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <ctype.h>
 
 /**
  * @return the name of the user from the conf file, or
  * an empty string if not found.  Must be freed by the caller
  */
 static inline char *malloc_username_from_conf_file()
 {
-    size_t len = 0;
-    char *buffer = malloc(1);
+    static const uint8_t BASE_LEN = 16;
+    size_t len = BASE_LEN;
+    char *buffer = malloc(BASE_LEN);
+    buffer[0] = '\0';
+
     FILE *fp = fopen("conf/username.txt","r");
-    ssize_t bytes_read = -1;
-    buffer[0] = 0;
     if ( fp != NULL ) {
         /**
          * See https://www.gnu.org/software/libc/manual/html_mono/libc.html#index-getdelim-994
          */
-        bytes_read = getline(&buffer, &len, fp);
+        ssize_t bytes_read = getline(&buffer, &len, fp);
```

Comment:
I see this link no longer works and probably links to the wrong place.  I think we could use something like https://man7.org/linux/man-pages/man3/getline.3.html instead.